src/Htmx/HtmxResponseHeaders.cs

```diff
@@ -194,38 +194,64 @@ public HtmxResponseHeaders Retarget(string value)
-        {
-            if (_headers.ContainsKey(Keys.Trigger))
-            {
-                throw new Exception("You must use either WithTrigger(..) or Trigger(..), but not both.");
-            }
+	    if (_triggers.ContainsKey(HtmxTriggerTiming.Default))
+	    {
+		    ParsePossibleExistingTriggers(Keys.Trigger, HtmxTriggerTiming.Default);
 
-            _headers[Keys.Trigger] = BuildTriggerHeader(HtmxTriggerTiming.Default);
-        }
+		    _headers[Keys.Trigger] = BuildTriggerHeader(HtmxTriggerTiming.Default);
+	    }
 
-        if (_triggers.ContainsKey(HtmxTriggerTiming.AfterSettle))
-        {
-            if (_headers.ContainsKey(Keys.TriggerAfterSettle))
-            {
-                throw new Exception("You must use either WithTrigger(..) or TriggerAfterSettle(..), but not both.");
-            }
+	    if (_triggers.ContainsKey(HtmxTriggerTiming.AfterSettle))
+	    {
+		    ParsePossibleExistingTriggers(Keys.TriggerAfterSettle, HtmxTriggerTiming.AfterSettle);
 
-            _headers[Keys.TriggerAfterSettle] = BuildTriggerHeader(HtmxTriggerTiming.AfterSettle);
-        }
+		    _headers[Keys.TriggerAfterSettle] = BuildTriggerHeader(HtmxTriggerTiming.AfterSettle);
+	    }
 
-        // ReSharper disable once InvertIf
-        if (_triggers.ContainsKey(HtmxTriggerTiming.AfterSwap))
-        {
-            if (_headers.ContainsKey(Keys.TriggerAfterSwap))
-            {
-                throw new Exception("You must use either WithTrigger(..) or TriggerAfterSwap(..), but not both.");
-            }
+	    // ReSharper disable once InvertIf
+	    if (_triggers.ContainsKey(HtmxTriggerTiming.AfterSwap))
+	    {
+		    ParsePossibleExistingTriggers(Keys.TriggerAfterSwap, HtmxTriggerTiming.AfterSwap);
 
-            _headers[Keys.TriggerAfterSwap] = BuildTriggerHeader(HtmxTriggerTiming.AfterSwap);
-        }
+		    _headers[Keys.TriggerAfterSwap] = BuildTriggerHeader(HtmxTriggerTiming.AfterSwap);
+	    }
 
-        return this;
+	    return this;
+    }
+
+    /// <summary>
+    /// Checks to see if the response has an existing header defined by headerKey.  If it does the
+    /// header loads all of the triggers locally so they aren't overwritten by Htmx.
+    /// </summary>
+    /// <param name="headerKey"></param>
+    /// <param name="timing"></param>
+    private void ParsePossibleExistingTriggers(string headerKey, HtmxTriggerTiming timing)
+    {
+	    if (!_headers.ContainsKey(headerKey))
+		    return;
+
+	    // Attempt to parse existing header as Json, if fails it is a simplified event key
+	    try
+	    {
+		    var existingTriggers = JsonSerializer.Deserialize<Dictionary<string, object?>>(_headers[headerKey]) ??
+		                           new Dictionary<string, object?>();
+
+		    // Load any existing triggers
+		    foreach (var eventName in existingTriggers.Keys)
+			    WithTrigger(eventName, existingTriggers[eventName], timing);
+	    }
+	    catch (System.Text.Json.JsonException)
```

Comment:
Is there a better way to do this rather than throwing and catching an exception? I feel like this will hurt performance in some way.

Reply:
```c#
try
{
    var reader = new Utf8JsonReader(Encoding.UTF8.GetBytes(_headers[headerKey]));
    // this will throw still :(
    var json = JsonNode.Parse(ref reader)?.AsObject();
    // Load any existing triggers
    foreach (var pair in json!)
        WithTrigger(pair.Key, pair.Value, timing);    
}
```

So this works, and we avoid the `Dictionary`. What do you think?